vizro-core/docs/pages/explanation/contributing.md

```diff
@@ -1,192 +1,159 @@
 
-Note that Hatch's `default` environment specifies `pre-commit` as a dependency but otherwise _does not_ specify dependencies for linting tools such as `black`. These are controlled by [.pre-commit-config.yaml](https://github.com/mckinsey/vizro/blob/main/.pre-commit-config.yaml) and can be updated when required with `pre-commit autoupdate`. Once per month, `pre-commit ci` raises a PR to do so.
+### `hatch run changelog:add`
 
-## Secret scans
+`hatch run changelog:add` generates a new changelog fragment. This is included in CI checks and required for all changes outside documentation.
 
-We use [gitleaks](https://github.com/gitleaks/gitleaks) for secret scanning. We do this via `pre-commit`, however there are a few things to note:
+The format of our changelog is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/). We use [scriv](https://pypi.org/project/scriv/) to build and maintain [our changelog](https://github.com/mckinsey/vizro/blob/main/vizro-core/CHANGELOG.md). When raising a PR, you must ensure that a changelog fragment has been created. This fragment is a small `.md` file describing your changes.
 
-1. Using `gitleaks` may require an installation of `go` on the developer machine. This is easy and explained in the [Go documentation](https://go.dev/doc/install).
-2. For that reason `hatch run lint` skips the secret scans, to function on all machines.
-3. To run a secret-scan, run `hatch run secrets`.
-4. Secret scans will run on CI, but it is highly recommended to check for secrets **before pushing to the remote repository** and ideally also before even committing.
+Run `hatch run changelog:add` to create a changelog fragment and then uncomment the relevant section(s). If you are uncertain about what to add or whether to add anything at all, refer to [Keep a Changelog](https://keepachangelog.com/en/1.0.0/). The rule of thumb is that if Vizro users would be affected in any way then the changes should be described in the changelog.
 
-When executing the secret scan, there are two modes: `protect` can discover secrets in staged files, `detect` does so in the commit history.
+!!! note
 
-## Changelog
+    Changes that only affect documentation do not need a changelog fragment. This facilitates simple modifications to documentation [made directly on GitHub](https://docs.github.com/en/repositories/working-with-files/managing-files/editing-files) or with the [github.dev](https://docs.github.com/en/codespaces/the-githubdev-web-based-editor), where no terminal is available to run `hatch changelog:add`. Any changes outside documentation require a changelog fragment to be generated. If your changes do not require a changelog entry then you still need to generate the fragment but can leave it all commented out.
```

Comment:
I have never found it to be the case that I can get away with no changelog, for any docs change. Is this a recent improvement?

Reply:
Not that recent: https://github.com/mckinsey/vizro/pull/480.

I did test it pretty thoroughly at the time and thought it seemed to work well, so if it's not working for you then please do point me towards a PR when you next find that it's not working as intended!